gremlin-go/driver/client.go

```diff
@@ -71,25 +73,51 @@ func NewClient(url string, configurations ...func(settings *ClientSettings)) (*C
 		transporterType: settings.TransporterType,
 		connection:      conn,
 	}
+	// TODO: PoolSize must be 1 on Session mode
 	return client, nil
 }
 
 // Close closes the client via connection.
 func (client *Client) Close() error {
-	return client.connection.close()
+	// If it is a Session, call closeSession
+	if client.Session != "" {
+		_, err := client.closeSession()
+		if err != nil {
+			return err
+		}
+	}
```

Comment:
Should this be an if - else?

Reply:
No, we just need to perform a few extra steps if it's a session

Reply:
Hmm then I guess the issue here is that if there is an error, the second part will not run despite it being a separate close.